app.jsx

```diff
@@ -11,69 +11,75 @@ import styled from "styled-components";
 
 import Colors from "./constants/colors";
 import Spacing from "./constants/spacing";
+import { BASE_STYLES, TAG_STYLES } from "./constants/styles";
 
-const parseNodes = (nodes, baseStyle = "normal") => {
+const getInlineStyle = (inlineStyle) => {
+    const isItalic = !!inlineStyle.match(/italic/);
+    const isBold = !!inlineStyle.match(/weight: 600|700|bold/);
+
+    let style = [];
+    if (isItalic) {
+        style = [...style, BASE_STYLES.ITALIC];
+    }
+    if (isBold) {
+        style = [...style, BASE_STYLES.BOLD];
+    }
+    
+    return style.length > 0 ? style : [];
+}
+
+const getStyle = ({name, attribs}) => {
+    const tagStyles = (name && TAG_STYLES[name] && [TAG_STYLES[name]]) || [];
+    const inlineStyles = (attribs && attribs.style && getInlineStyle(attribs.style)) || [];
+    const aggregatedStyles = [...tagStyles, ...inlineStyles];
+    return aggregatedStyles.length > 0 ? aggregatedStyles : [BASE_STYLES.NORMAL];
+}
+
+const parseNodes = (nodes, el) => {
     let parsed = [];
     for (const node of nodes) {
-        const { attribs, children, data, name } = node;
-        if (!name) {
+        const { children, data, parent } = node;
+
+        let elementStyle;
+        if (!el) {
+            elementStyle = new Set(getStyle(parent));
+        } else {
+            getStyle(parent).forEach(style => el.add(style));
+            elementStyle = el;
+        } 
+
+        if (data) {
+            if (elementStyle.has(BASE_STYLES.BOLD) || elementStyle.has(BASE_STYLES.ITALIC)) {
+                elementStyle.delete(BASE_STYLES.NORMAL);
+            };
```

Comment:
4. Once we're ready to output the json object, we want to remove a normal style coming from a div, if the element has a bold or italic styling.